File: src/IsbInterop/Tasks/ITaskFactory.vb
Imports IsbInterop.Base

Namespace Tasks

  ''' <summary>
  ''' Фабрика задач.
  ''' </summary>
  Public Interface ITaskFactory
    Inherits IEdmsObjectFactory(Of ITask, ITaskInfo)

    ''' <summary>
    ''' Создать новую задачу.
    ''' </summary>
    ''' <returns>Новая задача.</returns>
    Function CreateNew() As ITask
  End Interface
End Namespace

File: src/IsbInterop/Tasks/ITaskFactory.cs
using IsbInterop.Base;
using IsbInterop.Edms;

namespace IsbInterop.Tasks
{
  /// <summary>
  /// Фабрика задач.
  /// </summary>
  public interface ITaskFactory : IEdmsObjectFactory<ITask, ITaskInfo>
  {
    /// <summary>
    /// Создать новую задачу.
    /// </summary>
    /// <returns>Новая задача.</returns>
    ITask CreateNew();
  }
}
